### DISCUSSION TITLE
the loading indicator showing with message "please wait while the application is re-connecting

### DISCUSSION THREAD

**Discussion by Abdghali13** [flet-dev/flet#1168] [Q&A]
I have an issue with running the flet package on Flutter app ,Although the python script is running on the same port .
any help with that ?

Restarted application in 3,774ms.
flutter: Page build: page
flutter: Page theme: ThemeMode.system
flutter: Page fonts build: page
flutter: MeterialApp.router build: page
flutter: currentConfiguration
flutter: Page navigator build: page
flutter: _buildNavigator build
flutter: Page media build
flutter: New page size: Size(393.0, 852.0)
flutter: Page brightness did not change.
flutter: Route changed to: /
flutter: Registering web client with route: /
flutter: currentConfiguration
flutter: Connecting to Flet server http://localhost:8550...
flutter: Connecting to WebSocket server ws://localhost:8550/ws...
flutter: currentConfiguration
flutter: registerWebClientInternal
flutter: Page build: page
flutter: Page theme: ThemeMode.system
flutter: Page fonts build: page
flutter: MeterialApp.router build: page
flutter: Page navigator build: page
flutter: _buildNavigator build
flutter: Page size did not change.
flutter: Page brightness did not change.
[VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: SocketException: Connection refused (OS Error: Connection refused, errno = 61), address = localhost, port = 53374
#0      _NativeSocket.startConnect (dart:io-patch/socket_patch.dart:694:35)
#1      _RawSocket.startConnect (dart:io-patch/socket_patch.dart:1855:26)
#2      RawSocket.startConnect (dart:io-patch/socket_patch.dart:27:23)
#3      Socket._startConnect (dart:io-patch/socket_patch.dart:2078:22)
#4      Socket.startConnect (dart:io/socket.dart:763:21)
#5      _ConnectionTarget.connect (dart:_http/http_impl.dart:2466:20)
#6      _HttpClient._getConnection.connect (dart:_http/http_impl.dart:2906:12)
#7      _HttpClient._getConnection (dart:_http/http_impl.dart:2911:12)
#8      _HttpClient._openUrl (dart:_http/http_impl.dart:2766:12)
#9      _HttpClient.openUrl (dart:_http/http_impl.dart:2604:7)
#10     _WebSocketImpl.connect (dart:_http/websocket_impl.dart:1021:42)
#11     WebSocket.connect (dart:_http/websocket.dart:320:22)
#12     new<…>
flutter: WebSocket stream error WebSocketChannelException: WebSocketChannelException: SocketException: Connection refused (OS Error: Connection refused, errno = 61), address = localhost, port = 53374
flutter: WebSocket stream closed
flutter: Reconnect in 1 seconds
flutter: Page build: page
flutter: Page theme: ThemeMode.system
flutter: Page fonts build: page
flutter: MeterialApp.router build: page
flutter: Page navigator build: page
flutter: _buildNavigator build
flutter: Connecting to Flet server http://localhost:8550...
flutter: Connecting to WebSocket server ws://localhost:8550/ws...
flutter: registerWebClientInternal
[VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: SocketException: Connection refused (OS Error: Connection refused, errno = 61), address = localhost, port = 53391
#0      _NativeSocket.startConnect (dart:io-patch/socket_patch.dart:694:35)
#1      _RawSocket.startConnect (dart:io-patch/socket_patch.dart:1855:26)
#2      RawSocket.startConnect (dart:io-patch/socket_patch.dart:27:23)
#3      Socket._startConnect (dart:io-patch/socket_patch.dart:2078:22)
#4      Socket.startConnect (dart:io/socket.dart:763:21)
#5      _ConnectionTarget.connect (dart:_http/http_impl.dart:2466:20)
#6      _HttpClient._getConnection.connect (dart:_http/http_impl.dart:2906:12)
#7      _HttpClient._getConnection (dart:_http/http_impl.dart:2911:12)
#8      _HttpClient._openUrl (dart:_http/http_impl.dart:2766:12)
#9      _HttpClient.openUrl (dart:_http/http_impl.dart:2604:7)
#10     _WebSocketImpl.connect (dart:_http/websocket_impl.dart:1021:42)
#11     WebSocket.connect (dart:_http/websocket.dart:320:22)
#12     new<…>
flutter: WebSocket stream error WebSocketChannelException: WebSocketChannelException: SocketException: Connection refused (OS Error: Connection refused, errno = 61), address = localhost, port = 53391
flutter: WebSocket stream closed
flutter: Reconnect in 1 seconds
flutter: Connecting to Flet server http://localhost:8550...
flutter: Connecting to WebSocket server ws://localhost:8550/ws...
flutter: registerWebClientInternal
[VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: SocketException: Connection refused (OS Error: Connection refused, errno = 61), address = localhost, port = 53407
#0      _NativeSocket.startConnect (dart:io-patch/socket_patch.dart:694:35)
#1      _RawSocket.startConnect (dart:io-patch/socket_patch.dart:1855:26)
#2      RawSocket.startConnect (dart:io-patch/socket_patch.dart:27:23)
#3      Socket._startConnect (dart:io-patch/socket_patch.dart:2078:22)
#4      Socket.startConnect (dart:io/socket.dart:763:21)
#5      _ConnectionTarget.connect (dart:_http/http_impl.dart:2466:20)
#6      _HttpClient._getConnection.connect (dart:_http/http_impl.dart:2906:12)
#7      _HttpClient._getConnection (dart:_http/http_impl.dart:2911:12)
#8      _HttpClient._openUrl (dart:_http/http_impl.dart:2766:12)
#9      _HttpClient.openUrl (dart:_http/http_impl.dart:2604:7)
#10     _WebSocketImpl.connect (dart:_http/websocket_impl.dart:1021:42)
#11     WebSocket.connect (dart:_http/websocket.dart:320:22)
#12     new<…>
flutter: WebSocket stream error WebSocketChannelException: WebSocketChannelException: SocketException: Connection refused (OS Error: Connection refused, errno = 61), address = localhost, port = 53407
flutter: WebSocket stream closed
flutter: Reconnect in 1 seconds
flutter: Connecting to Flet server http://localhost:8550...
flutter: Connecting to WebSocket server ws://localhost:8550/ws...
flutter: registerWebClientInternal

**Comment by moluopro** [comment]
Me too !!! And the code in the documentation of https://pub.dev/packages/flet is also obsolete, the constructor of the FletApp class is missing the assetsDir parameter.

**Comment by FeodorFitsner** [comment]
Are you able to open http://localhost:8550 in the browser?

**Comment by larsien** [comment]
is it resolved? I encountered same issue. cannot run sample app in flutter with that error message

**Comment by larsien** [comment]
> Are you able to open http://localhost:8550 in the browser?

I tried to connect to "http://localhost:8550" on browser in android/web environment. 
but it failed to connect both

**Reply by larsien** [reply]
and this is my error log 

E/flutter ( 7429): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = localhost, port = 39626
E/flutter ( 7429): #0      _NativeSocket.startConnect (dart:io-patch/socket_patch.dart:719:35)
E/flutter ( 7429): #1      _RawSocket.startConnect (dart:io-patch/socket_patch.dart:1877:26)
E/flutter ( 7429): #2      RawSocket.startConnect (dart:io-patch/socket_patch.dart:27:23)
E/flutter ( 7429): #3      Socket._startConnect (dart:io-patch/socket_patch.dart:2100:22)
E/flutter ( 7429): #4      Socket.startConnect (dart:io/socket.dart:766:21)
E/flutter ( 7429): #5      _ConnectionTarget.connect (dart:_http/http_impl.dart:2486:20)
E/flutter ( 7429): #6      _HttpClient._getConnection.connect (dart:_http/http_impl.dart:2924:12)
E/flutter ( 7429): #7      _HttpClient._getConnection (dart:_http/http_impl.dart:2929:12)
E/flutter ( 7429): #8      _HttpClient._openUrl (dart:_http/http_impl.dart:2784:12)
E/flutter ( 7429): #9      _HttpClient.openUrl (dart:_http/http_impl.dart:2622:7)
E/flutter ( 7429): #10     _WebSocketImpl.connect (dart:_http/websocket_impl.dart:1021:42)
E/flutter ( 7429): #11     WebSocket.connect (dart:_http/websocket.dart:320:22)
E/flutter ( 7429): #12     new IOWebSocketChannel.connect (package:web_socket_channel/io.dart:81:28)
E/flutter ( 7429): #13     connect (package:web_socket_channel/src/_connect_io.dart:15:24)
E/flutter ( 7429): #14     new WebSocketChannel.connect (package:web_socket_channel/src/channel.dart:115:16)
E/flutter ( 7429): #15     FletWebSocketServerProtocol.connect (package:flet/src/flet_server_protocol_web_socket.dart:40:35)
E/flutter ( 7429): <asynchronous suspension>
E/flutter ( 7429): #16     FletServer.connect (package:flet/src/flet_server.dart:65:7)
E/flutter ( 7429): <asynchronous suspension>
E/flutter ( 7429): #17     FletServer._onDisconnect.<anonymous closure> (package:flet/src/flet_server.dart:100:9)
